src/Microsoft.Health.Fhir.CodeGen/CompareTool/PackageComparer.cs

```diff
@@ -1402,6 +1413,244 @@ private void WriteValueSetMaps(string outputDir, IEnumerable<ValueSetComparison>
+                {
+                    var group = new StructureMap.GroupComponent();
+                    group.Name = String.Join("", elementComparison.Source.Path.Split('.').Select(p => p.Substring(0, 1).ToUpper() + p.Substring(1))); // need to convert the path into a better name than this so you get AccountGuarantor
+                    group.Input.Add(new StructureMap.InputComponent()
+                    {
+                        Mode = StructureMap.StructureMapInputMode.Source,
+                        Name = "src",
+                        // Type = elementComparison.Source.Types
+                    });
+                    group.Input.Add(new StructureMap.InputComponent()
+                    {
+                        Mode = StructureMap.StructureMapInputMode.Target,
+                        Name = "tgt",
+                        // Type = elementComparison.Source.Types
+                    });
+                    if (!elementComparison.Source.Path.Contains("."))
+                    {
+                        group.Input[0].Type = srcResourceType + srcVersion.ToRLiteral();
+                        group.Input[1].Type = tgtResourceType + tgtVersion.ToRLiteral();
+                        group.Extends = "Resource";
+                    }
+                    if (elementComparison.Source.Types.Any(t => t.Value.Name == "BackboneElement"))
+                    {
+                        group.Extends = "BackboneElement";
+                    }
+                    if (!elementComparison.Source.Types.Any() && !elementComparison.Source.Path.Contains("."))
+                    {
+                        // This is the root type mapping, so can be the default
+                        group.TypeMode = StructureMap.StructureMapGroupTypeMode.TypeAndTypes;
+                    }
+                    sm.Group.Add(group);
+
+                    // Now scan for the properties at this level in the resource
+                    foreach ((string pathInner, ElementComparison elementComparisonInner) in propComparisons)
+                    {
+                        if (!elementComparisonInner.Source.Path.Contains("."))
+                            continue;
+                        if (elementComparisonInner.Source.Types.Any(t => t.Value.Name == "BackboneElement"))
+                        {
+                            // This is the calling of the backbone element copying part
+                            // continue;
+                        }
+
+                        if (!elementComparisonInner.Source.Path.StartsWith($"{path}.") || elementComparisonInner.Source.Path.LastIndexOf(".") > elementComparison.Source.Path.Length)
+                            continue;
+
+                        // Filter out any backbone element props
+                        var backboneElementProps = new[] { "id", "extension", "modifierExtension" };
```

Comment:
There has to be a better way to locate if the property isn't from the differential (and thus not required at this level in the resource)